src/datasets/load.py

```diff
@@ -1308,6 +1314,130 @@ def get_module(self) -> DatasetModule:
+                    )
+        except Exception as e:  # noqa catch any exception of the datasets-server and consider the parquet export doesn't exist
+            logger.debug(f"No parquet export for {self.name} available ({type(e).__name__}: {e})")
+        raise DatasetsServerError("No exported Parquet files available.")
+
+    def _get_exported_dataset_infos(self) -> Dict[str, Dict[str, Any]]:
+        datasets_server_info_url = config.HF_ENDPOINT.replace("://", "://datasets-server.") + "/info?dataset="
+        try:
+            info_response = http_get(
+                url=datasets_server_info_url + self.name,
+                temp_file=None,
+                headers=get_authentication_headers_for_url(config.HF_ENDPOINT + f"datasets/{self.name}"),
+                timeout=100.0,
+                max_retries=3,
+            )
+            info_response.raise_for_status()
+            if "X-Revision" in info_response.headers:
+                if info_response.headers["X-Revision"] == self.from_sha or self.from_sha is None:
+                    info_response = info_response.json()
+                    if (
+                        info_response.get("partial") is False
+                        and not info_response.get("pending", True)
+                        and not info_response.get("failed", True)
+                        and "dataset_info" in info_response
+                    ):
+                        return info_response["dataset_info"]
+                    else:
+                        logger.debug(f"Dataset info for {self.name} is not completely ready yet.")
+                else:
+                    logger.debug(
+                        f"Dataset info for {self.name} is available but outdated (revision='{info_response.headers['X-Revision']}')"
+                    )
+        except Exception as e:  # noqa catch any exception of the datasets-server and consider the dataset info doesn't exist
+            logger.debug(f"No dataset info for {self.name} available ({type(e).__name__}: {e})")
+        raise DatasetsServerError("No exported dataset infos available.")
+
+    def get_module(self) -> DatasetModule:
+        exported_parquet_files = self._get_exported_parquet_files()
+        exported_dataset_infos = self._get_exported_dataset_infos()
+        hfh_dataset_info = HfApi(config.HF_ENDPOINT).dataset_info(
+            self.name,
+            revision="refs/convert/parquet",
+            token=self.download_config.token,
+            timeout=100.0,
+        )
+        # even if metadata_configs is not None (which means that we will resolve files for each config later)
+        # we cannot skip resolving all files because we need to infer module name by files extensions
+        revision = hfh_dataset_info.sha  # fix the revision in case there are new commits in the meantime
+        metadata_configs = MetadataConfigs._from_exported_parquet_files(
```

Comment:
We should be able to avoid using MetadataConfigs for this...

But this can be addressed later (this YAML metadata API is a mess, so it needs a bigger refactor).